Module validation

PonkinDmitrij2 opened the pull request.

PonkinDmitrij2 committed:
Adds implementation of the validation module

PonkinDmitrij2 committed:
Changes the implementation of the isValid method

PonkinDmitrij2 committed:
Merge branch 'master' of github.com:PonkinDmitrij2/cookie-hub into module-validation

PonkinDmitrij2 committed:
Adds a validation for the login form

PonkinDmitrij2 committed:
Merge branch 'master' of github.com:PonkinDmitrij2/cookie-hub into module-validation

PonkinDmitrij2 committed:
Changes the implementation of the validation module

PonkinDmitrij2 committed:
Optimizes the isValid method

Transfers the destructive assignment to the for of loop initialization

Kamys commented on src/validation/index.js:
```diff
@@ -24,18 +27,21 @@ const isValid = (form, schemaValidation) => {
-      if (param[0] === 'custom') {
-        if (param[1](fieldValue, form)) {
-          continue;
+      const [methodName, methodArg] = param;
```
Наименование не очень правильное. Переменные methodName и methodArg могут не всегда содержать метод. Нужно что то типа ruleName и ruleArgs.

Kamys commented on src/validation/index.js:
```diff
@@ -11,6 +11,9 @@ const validationMethods = {
   regex(value, regex) {
     return regex.test(value);
   },
+  customCall(value, form, fn) {
```
Смотри можно метод customCall назвать как custom и тогда не нужно будет делать проверку на if в 33 строке кода.

PonkinDmitrij2 committed:
Changes variable names in isValid method

Changes the names of the variables "methodName" and "methodArg" to
"ruleName" and "ruleArgs" to more accurately reflect the essence.

PonkinDmitrij2 committed:
Changes the validation module

Changes module logic when working with custom methods.
